WR397535: Make shortname filter cross-DB compatible

marxjohnson opened the pull request:
Switch `~` for `$DB->sql_regex()` in the query used by local_ace_get_enrolled_courses(). This allows it to work on MySQL as well as PostgreSQL.

danmarsden commented:
thanks @marxjohnson - couple of codechecker issues that it's reporting there - any chance you could take a look at those?

nice to see the new tests too :-)

marxjohnson commented:
Thanks Dan, I've updated the commit to fix the issues in the unit test file, and added an additional commit fixing the issues elsewhere which aren't related to this change.

danmarsden commented:
Thanks @marxjohnson - I note this changes the postgres regex expression from using a "~" to now using "~*" - which I don't think is an issue, but I'll do some searching to see if there are any performance impacts (negative or positive) between the 2 options.

danmarsden commented:
yeah - can't find any reason or reports of the case-insensitive version being slower than the other one and in our current use we're using a case-insensitve regex anyway so I'm just going to merge this in - thanks!
